homeassistant/components/ialarmxr/config_flow.py

```diff
@@ -0,0 +1,75 @@
+"""Config flow for Antifurto365 iAlarmXR integration."""
+import logging
+from logging import Logger
+
+from pyialarmxr import IAlarmXR, IAlarmXRGenericException
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DEFAULT_HOST, DEFAULT_PORT, DOMAIN
+
+_LOGGER: Logger = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=DEFAULT_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def _get_device_mac(
+    hass: core.HomeAssistant, host: str, username: str, password: str, port: int
+) -> str:
+    ialarmxr = IAlarmXR(host, username, password, port)
+    return await hass.async_add_executor_job(ialarmxr.get_mac)
+
+
+class IAlarmConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Antifurto365 iAlarmXR."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+        mac = None
+
+        if user_input is None:
```

Comment:
If you reverse the condition you can drop the duplicate `async_show_form` and let the branch fall through to end of the function with errors set before calling `async_show_form`